src/main/java/de/tum/cit/aet/artemis/lecture/service/LectureService.java

```diff
@@ -191,4 +213,58 @@ public void ingestTranscriptionInPyris(LectureTranscription transcription, Cours
     public void deleteLectureTranscriptionInPyris(LectureTranscription existingLectureTranscription) {
         irisLectureApi.ifPresent(webhookService -> webhookService.deleteLectureTranscription(existingLectureTranscription));
     }
+
+    public Lecture getForDetails(long lectureId, User user) {
+        Lecture lecture = lectureRepository.findByIdWithLectureUnitsAndAttachmentsElseThrow(lectureId);
+        Course course = lecture.getCourse();
+        if (course == null) {
+            throw new BadRequestAlertException("The course belonging to this lecture does not exist", "lecture", "courseNotFound");
+        }
+        Set<LectureUnitCompletion> completionsForLectureAndUser = lectureUnitRepository.findCompletionsForLectureAndUser(lectureId, user.getId());
+        Map<Long, LectureUnitCompletion> byUnit = completionsForLectureAndUser.stream().collect(toMap(cu -> cu.getLectureUnit().getId(), cu -> cu));
+
+        lecture.getLectureUnits().forEach(lu -> {
+            Set<LectureUnitCompletion> completions = new HashSet<>();
+            if (byUnit.containsKey(lu.getId())) {
+                completions.add(byUnit.get(lu.getId()));
+            }
+            lu.setCompletedUsers(completions);
+        });
+        competencyApi.ifPresent(api -> api.addCompetencyLinksToExerciseUnits(lecture));
+        return filterLectureContentForUser(lecture, user);
+
+    }
+
+    private Lecture filterLectureContentForUser(Lecture lecture, User user) {
+        lecture = filterActiveAttachments(lecture, user);
+
+        // The Objects::nonNull is needed here because the relationship lecture -> lecture units is ordered and
+        // hibernate sometimes adds nulls into the list of lecture units to keep the order
+        Set<Exercise> relatedExercises = lecture.getLectureUnits().stream().filter(Objects::nonNull).filter(lectureUnit -> lectureUnit instanceof ExerciseUnit)
+                .map(lectureUnit -> ((ExerciseUnit) lectureUnit).getExercise()).collect(Collectors.toSet());
```

Comment:
(Optional suggestion): For consistency with your similar change in `CompetencyService`:
```suggestion
        Set<Exercise> relatedExercises = lecture.getLectureUnits().stream().filter(Objects::nonNull).filter(ExerciseUnit.class::isInstance)
                .map(ExerciseUnit.class::cast).map(ExerciseUnit::getExercise).collect(Collectors.toSet());
```